samza-core/src/test/java/org/apache/samza/execution/TestStreamEdge.java

```diff
@@ -50,8 +50,9 @@ public void testGetStreamSpec_Batch() {
     Map<String, String> config = new HashMap<>();
     config.put(ApplicationConfig.APP_MODE, ApplicationConfig.ApplicationMode.BATCH.name());
     config.put(ApplicationConfig.APP_RUN_ID, "123");
-    StreamEdge edge = new StreamEdge(spec, true, false, new MapConfig(config));
-    assertEquals(edge.getStreamSpec().getPhysicalName(), spec.getPhysicalName() + "-123");
+    StreamSpec batchSpec = new StreamSpec("stream-1", "stream-1", "system-1");
+    StreamEdge edge = new StreamEdge(batchSpec, true, false, new MapConfig(config));
+    assertEquals(edge.getStreamSpec().getPhysicalName(), batchSpec.getPhysicalName() + "-123");
```

Comment:
shouldn't you have another test case where the run.id is already applied and you verify it's not re-applied?

Reply:
+1, could there also add a test such that this does not trigger for intermediate streams for nonbatch?